src/unix.rs

```diff
@@ -0,0 +1,32 @@
+extern crate libc;
+
+use std::os::unix::io::RawFd;
+use std::io;
+use std::mem;
+use Dims;
+use Result;
+
+/// Get the window size from a file descriptor (0 for stdout)
+///
+/// Option is returned as there is no distinction between errors (all ENOSYS)
+pub fn win_size(fd: RawFd) -> Result<libc::winsize> {
+    unsafe {
+        let ws: libc::winsize = mem::uninitialized();
+        if libc::ioctl(fd, libc::TIOCGWINSZ, &ws) == 0 {
+            Ok(ws)
+        } else {
+            Err(io::Error::last_os_error().into())
+        }
+    }
+}
+
+impl From<libc::winsize> for Dims {
```

Comment:
Any reason to implement this? I can't see users actually needing to call this outside of term. I'd just do this conversion directly (i.e., implement a function or inline the conversion) instead of implementing `From`.